src/NHibernate/Engine/Query/QueryPlanCache.cs

```diff
@@ -157,16 +157,15 @@ private ParameterMetadata BuildNativeSQLParameterMetadata(string sqlString)
 				string name = entry.Key;
 				ParamLocationRecognizer.NamedParameterDescription description = entry.Value;
 				namedParamDescriptorMap[name] =
-					new NamedParameterDescriptor(name, null, description.JpaStyle);				
+					new NamedParameterDescriptor(name, null, description.JpaStyle);
 			}
 
 			return new ParameterMetadata(ordinalDescriptors, namedParamDescriptorMap);
 		}
 
 		[Serializable]
-		private class HQLQueryPlanKey : IEquatable<HQLQueryPlanKey>
+		private class HQLQueryPlanKey : QueryPlanKey
```

Comment:
`IEquatable` was having no usage here because the class was stored in a non generic dictionary. Furthermore, now it is stored in a generic dictionary of `QueryPlanKey`, which implements `IEquatable<QueryPlanKey>`. I am wrong believing it is useless to add `IEquatable` for subclasses not stored in collection specifically using their type, not the base one? Adding the specialized `IEquatable` for subclasses along with base equatable implementation is incurring some code duplication or inefficiencies at least.